src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -75,23 +63,39 @@ public class MultiSearchService implements CommitListener {
 	
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 	
-	Map<String, String> publishedBranches = new HashMap<>();
-	
-	BoolQueryBuilder cachedBranchesQuery = null;
-	LocalDate cacheDate = null;
+	private final Map<String, String> publishedBranches = new HashMap<>();
 
-	public Page<Description> findDescriptions(DescriptionCriteria criteria, PageRequest pageRequest) {
-		
-		SearchHits<Description> searchHits = findDescriptionsHelper(criteria, pageRequest);
+	private MultiBranchCriteria cachedBranchCriteria = null;
+	private LocalDate cacheDate = null;
+
+	public Page<Description> findDescriptions(DescriptionCriteria criteria, String ecl, PageRequest pageRequest, Set<String> includeBranches) {
+
+		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",getBranchesQuery().getTimepoint(),new ArrayList<>(getBranchesQuery().getBranchCriteria()));
```

Comment:
@RichardAWood, `getBranchesQuery` is called twice on this line. Unless the response is expected to be different between the first and second call, then the response could be stored in a variable. This should help with making the search's performance.

Reply:
I've gone ahead and saved the getBranchesQuery output to a variable as suggested.